src/hotspot/os/linux/osContainer_linux.cpp

```diff
@@ -60,15 +58,8 @@ void OSContainer::init() {
   if (cgroup_subsystem == NULL) {
     return; // Required subsystem files not found or other error
   }
-  // We need to update the amount of physical memory now that
-  // cgroup subsystem files have been processed.
-  if ((mem_limit = cgroup_subsystem->memory_limit_in_bytes()) > 0) {
```

Comment:
I'm concerned that this removal of the cgroup lookup re-introduces part of https://bugs.openjdk.org/browse/JDK-8232207

Without this call in early init we could potentially lose some of the caching effects we previously had.

Reply:
Hi Severin, thanks for your comment. As I understand [/JDK-8232207](https://bugs.openjdk.org/browse/JDK-8232207), a 20ms delay cache was added to `OSContainer::memory_limit_in_bytes`, and `OSContainer::init` called that routine, starting the 20ms timer. Any routines also calling `OSContainer::memory_limit_in_bytes` within 20ms of `OSContainer::init` would benefit from the caching.

This PR removes that cache seeding in `OSContainer::init`, so any caller to `OSContainer::memory_limit_in_bytes` within 20ms of `init` will indeed be slower. But that caller will seed the cache, so subsequent calls (within 20ms) will be unaffected. And, we're just moving the initial delay from `OSContainer::init` to the next caller: overall the delay will be the same.

Furthermore, in the event that there wasn't another caller to `memory_limit_in_bytes` within 20ms of `OSContainer::init`, the PR will speed things up, as prior, the cache is being seeded unnecessarily.

I am happy to try and do some performance comparisons if required.